x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/repositories/PutInternalRepositoryRequest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ccr.action.repositories;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Settings;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class PutInternalRepositoryRequest extends ActionRequest {
+
+    private final String name;
+    private final String type;
+    private final Settings settings;
+
+    public PutInternalRepositoryRequest(String name, String type) {
+        this(name, type, Settings.EMPTY);
+    }
+
+    public PutInternalRepositoryRequest(String name, String type, Settings settings) {
```

Comment:
If we rename this to `PutInternalCCRRepositoryRequest`, we can get rid of the settings parameter here. We don't use it for CCR.